packages/deployed-backend-client/src/deployed_backend_client.ts

```diff
@@ -134,12 +136,67 @@ export class DefaultDeployedBackendClient implements DeployedBackendClient {
     };
   };
 
+  /**
+   * Returns Amplify stacks which failed to delete for the account and region
+   */
+  listDeleteFailedStacks =
```

Comment:
This file can use some refactoring. We are duplicating the entire logic as the `listSandboxes`.

We should create one parameterized method that generates the stackMetaData based on inputs and the client (console) can pass the relevant values (e.g. a status filter). We don't need to create new APIs for each filter or type of deployment.

Reply:
hard +1. My suggestion would be to refactor this method and `listSandboxes` into a single `listBackends` method that has the filter args that we want (structured in a way that we can add additional types of filters as the need arises).

We have also taken the stance that the API of this package should not leak the "stack" concept. Here we are leaking both stack and stack status (listDeleteFailedStack).
FWIW, we've already broken that rule a little bit with the `DeployedBackendResource` type, but that doesn't mean we should continue leaking more.

Reply:
Refactored into a single method